Source/Utils/WarpXUtil.cpp

```diff
@@ -239,6 +271,30 @@ void Store_parserString(const amrex::ParmParse& pp, std::string query_string,
     f.clear();
 }
 
+int safeCastToInt(amrex::Real x, const std::string& real_name) {
+    int result = 0;
+    bool error_detected = false;
+    std::string assert_msg;
+    // (2.0*(numeric_limits<int>::max()/2+1)) converts numeric_limits<int>::max()+1 to a real ensuring accuracy to all digits
+    // This accepts x = 2**31-1 but rejects 2**31.
+    if (x < (2.0*(std::numeric_limits<int>::max()/2+1))) {
+        if (std::ceil(x) >= std::numeric_limits<int>::min()) {
+            result = static_cast<int>(x);
+        } else {
+            error_detected = true;
+            assert_msg = "Error: Negative overflow detected when casting " + real_name + " = " + std::to_string(x) + " to int";
+        }
+    } else if (x > 0) {
```

Comment:
I am a bit confused why `x>0` is an overflow error - do we need a doxygen string to explain the value range that we put in here?

Reply:
Because of the first `if` test, if `x > 0` this means that `x > std::numeric_limits<int>::max()`, hence an overflow. If `x > 0` is false, this means that x is NaN since all comparisons to a NaN return false.

Reply:
Thanks a lot :)